lib/rails-settings/cached_settings.rb

```diff
@@ -15,7 +15,12 @@ def cache_key
     class << self
       def cache_key(var_name, scope_object)
         scope = "rails_settings_cached:"
-        scope << "#{scope_object.class.base_class.to_s}-#{scope_object.id}:" if scope_object
+        if scope_object
+          klass = scope_object.class
+          primary_key = klass.primary_key
+          id = scope_object.public_send(primary_key)
```

Comment:
And the `object.id` it will use primary key

```rb
def id
  attr_name = self.class.primary_key
  column = column_for_attribute(attr_name)

  self.class.send(:define_read_method, :id, attr_name, column)
  # now that the method exists, call it
  self.send attr_name.to_sym
end
```